### ISSUE TITLE
[BUG] Segmentation fault in `queue.c`

### ISSUE THREAD

**Issue by Arash1381-y** [TheAlgorithms/C#1269]
### Description

In the queue model (path : data_structures/queue/queue.c). dequeue won't work due to a simple bug on the last if clause of it. take a look:
https://github.com/TheAlgorithms/C/blob/b1a8da69a87cd45e9db0cc340730e4a5e3b36201/data_structures/queue/queue.c#L65

as you can see if we have only one variable in our queue on dequeue the HEAD will be null and assigning NULL to NULL->next will cause a segmentation fault.

### Expected behavior

not to cause fault =)

### Actual behavior

case segmentation fault

### Possible fix

simply put line 65 in its above clause

### Steps to reproduce

1. call create
2. enqueue anything you like 
3. dequeue it 

### Context

It just don't work =)
